feat: add organization setting to control public sharing

zlwaterfield opened the pull request:
## Changes

Introduces the allow_publicly_shared_resources field to Organization, enabling admins to control whether resources can be shared publicly. Updates frontend to respect this setting in sharing modals and organization settings, and enforces the restriction in backend sharing endpoints. Existing sharing links and public dashboards are blocked if the setting is disabled.

## How did you test this code?

- Tests

<!-- Docs reminder: If this change requires updated docs, please do that! Engineers are the primary people responsible for their documentation. 🙌 -->

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

zlwaterfield committed:
Add organization setting to control public sharing

Introduces the allow_publicly_shared_resources field to Organization, enabling admins to control whether resources can be shared publicly. Updates frontend to respect this setting in sharing modals and organization settings, and enforces the restriction in backend sharing endpoints. Existing sharing links and public dashboards are blocked if the setting is disabled.

yasen-posthog committed:
Add migrations

yasen-posthog committed:
Fixups on FE.

yasen-posthog committed:
Separate the org security settings into a standalone section.

yasen-posthog committed:
Add a test to validate if the sharing restriction works

github-actions[bot] committed:
Update UI snapshots for `chromium` (14)

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

yasen-posthog committed:
Fix tests

yasen-posthog committed:
Add a feature check before enforcing the restriction.

yasen-posthog committed:
Amend tests

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update query snapshots

yasen-posthog committed:
Update snap

zlwaterfield committed:
Merge branch 'master' into zach/org-sharing-settings

zlwaterfield committed:
Show warning banner when sharing is disabled

Added a LemonBanner to inform users when public sharing is disabled for the organization. Simplified the sharing switch label and ensured sharing options are only available when allowed.

a-lider commented on frontend/src/scenes/settings/SettingsMap.tsx:
```diff
@@ -681,6 +682,18 @@ export const SETTINGS_MAP: SettingSection[] = [
+        title: 'Security settings',
+        settings: [
+            {
+                id: 'organization-proxy',
```
Legit comment

zlwaterfield committed:
Update security settings section ID and types

Changed the settings section ID from 'organization-proxy' to 'organization-security' in SettingsMap. Added 'organization-settings' to SettingSectionId and SettingId types for improved type coverage.

zlwaterfield committed:
Update organization settings types

Removed 'organization-settings' from SettingSectionId and SettingId types, and adjusted the order of 'organization-danger-zone' and 'organization-security'. This streamlines the available organization settings options.

a-lider reviewed (approved):
Tested locally, looks good

zlwaterfield committed:
Add allow_publicly_shared_resources to org insert

Updated the test utility to include the allow_publicly_shared_resources field when inserting a new organization into the database. This ensures test data matches the latest schema.

zlwaterfield committed:
Update test_utils.rs

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/org-sharing-settings

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/org-sharing-settings

pauldambra reviewed (approved):
👍

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

a-lider committed:
Merge branch 'master' into zach/org-sharing-settings

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/org-sharing-settings

zlwaterfield committed:
Merge branch 'master' into zach/org-sharing-settings

zlwaterfield committed:
Merge remote-tracking branch 'origin/master' into zach/org-sharing-settings

zlwaterfield committed:
Update migration dependency and rename migration file

Renamed migration file to 0820_organization_allow_publicly_shared_resources.py and updated its dependency to reference migration 0819_alter_integration_kind. This ensures the migration order is correct.

github-actions[bot] committed:
Update UI snapshots for `chromium` (11)
